### ISSUE TITLE
OpenFGA and PgBouncer - is this a good combo?

### ISSUE THREAD

**Issue by filipstiglic** [openfga/openfga#479]
Hello there :)

First of all, thank you very much for this project and for your effort :) And now the question(our main goal is to make OpenFGA as fast, as possible): **Can openFGA benefit from pgbouncer(connection pooler)?** I'm asking this, because I'm not absolutely sure - because openFGA has a setting(among others) which specifies number of idle database connections, so that kind of implicates, that some connection pooling is already happening inside openFGA and if that's the case, additional connection pooler won't probably be necessary - but I'm not sure.

Thnak you very much for explaining, have a really nice day.

**Comment by rhamzeh** [comment]
Thanks for asking @filipstiglic - right now we don't have enough tests to fully confirm. However we are investigating and working on [several issues](https://github.com/orgs/openfga/discussions/98) impacting performance with PostgreSQL, some related to dropped connections and the way we are doing our pooling.

We've made some progress and are working on continuing that route. PG Bouncer is something which we considered and we still have it as "to investigate" to see if it does result in performance improvements or just increased latency for little or no bump in scalability when it comes to DB connections.

So TLDR: We don't know yet, but we plan on investigating PG Bouncer as one of the many performance and scalability improvements that can be brought to OpenFGA in a prod env.

**Comment by filipstiglic** [comment]
> Thanks for asking @filipstiglic - right now we don't have enough tests to fully confirm. However we are investigating and working on [several issues](https://github.com/orgs/openfga/discussions/98) impacting performance with PostgreSQL, some related to dropped connections and the way we are doing our pooling.
> 
> We've made some progress and are working on continuing that route. PG Bouncer is something which we considered and we still have it as "to investigate" to see if it does result in performance improvements or just increased latency for little or no bump in scalability when it comes to DB connections.
> 
> So TLDR: We don't know yet, but we plan on investigating PG Bouncer as one of the many performance and scalability improvements that can be brought to OpenFGA in a prod env.

Thank you very much for the answer :) I've seen some improvements regarding pgsql connections, already present in the "main" of 0.3.3 , so looking forward to the release :)

**Comment by aaguiarz** [comment]
Hi @filipstiglic 

Have you been able to successfully utilize OpenFGA? If so, I would love to hear about your experience and any feedback you may have.

If you don't mind sharing, please email me directly at andres.aguiar at openfga.dev

Thanks!
